### PR TITLE
Add isHeadless capability

### PR DESCRIPTION
**Pull request by vrunoa**
Add isHeadless capability to make the emulator to launch headless.

### CODE REVIEW

File: test/unit/android-helper-specs.js [hunk 1]
```diff
@@ -80,6 +78,15 @@ describe('Android Helpers', () => {
     it('should fail if avd name is not specified', async () => {
       await helpers.prepareEmulator(adb, {}).should.eventually.be.rejected;
     });
+    it('should set the correct avdArgs', async () => {
```
Comment (imurchie) [inline test/unit/android-helper-specs.js]: This should be in a separate `describe` block for `prepareAVDArgs`, not tacked onto the one for `prepareEmulator`.

File: lib/android-helpers.js [hunk 2]
```diff
@@ -60,12 +60,20 @@ helpers.prepareEmulator = async function (adb, opts) {
...
+
+helpers.prepareAVDArgs = function (opts, adb, avdArgs) {
   if (!_.isUndefined(opts.networkSpeed)) {
     let networkSpeed = this.ensureNetworkSpeed(adb, opts.networkSpeed);
     avdArgs += ` -netspeed ${networkSpeed}`;
   }
-  await adb.launchAVD(avd, avdArgs, language, locale, avdLaunchTimeout,
-                      avdReadyTimeout);
+  if (!_.isUndefined(opts.isHeadless) && opts.isHeadless === true) {
+    avdArgs += ' -no-window';
```
Comment (mykola-mokhnach) [inline lib/android-helpers.js]: it looks a bit redundant to me that we add strings leading with spaces. I'd rather create an array of additional args and push values there and then do something like avdArgs + ` ${array.join(' ')}` at the end.

File: lib/android-helpers.js [hunk 3]
```diff
@@ -60,12 +60,20 @@ helpers.prepareEmulator = async function (adb, opts) {
...
+};
+
+helpers.prepareAVDArgs = function (opts, adb, avdArgs) {
   if (!_.isUndefined(opts.networkSpeed)) {
     let networkSpeed = this.ensureNetworkSpeed(adb, opts.networkSpeed);
     avdArgs += ` -netspeed ${networkSpeed}`;
   }
-  await adb.launchAVD(avd, avdArgs, language, locale, avdLaunchTimeout,
-                      avdReadyTimeout);
+  if (!_.isUndefined(opts.isHeadless) && opts.isHeadless === true) {
```
Comment (mykola-mokhnach) [inline lib/android-helpers.js]: it would be enough here to just verify `if (opts.isHeadless)`

**Comment by mykola-mokhnach** [comment]
should we document the new capability somewhere?

**Comment by vrunoa** [comment]
@mykola-mokhnach yes. I'll add a PR on appium

**Comment by imurchie** [comment]
Published in `appium-android-driver@1.25.2`.

**Comment by imurchie** [comment]
@vrunoa There is no cap list on this repo. This should go in the general list on Appium, since it's now available on iOS (Xcode 9+) and Android.

**Comment by vrunoa** [comment]
@imurchie https://github.com/appium/appium/pull/9064

**Pull request merged**